src/odemis/gui/cont/tabs.py

```diff
@@ -3543,6 +3554,10 @@ def _onAlignMode(self, mode):
         if mode != "fiber-align" and main.spec_sel:
             main.spec_sel.position.unsubscribe(self._onFiberPos)
 
+        if mode != "fiber-align":
+            # Unsubscribe from data of photo-detector 2
+            if self._photods and len(self._photods) > 1:
```

Comment:
I'd rather have an explicit `self._fbdet2` variable (which is None if there is no detector), so that it's possible to just write `if self._fbdet2`.